src/site/antora/modules/ROOT/pages/manual/migration.adoc

```diff
@@ -16,433 +16,573 @@ Licensed to the Apache Software Foundation (ASF) under one or more
+. Set the `CLASSPATH` environment variable to contain the artifacts mentioned above.
+. Run
 
-=== Layouts
+[source,shell]
+----
+java org.apache.log4j.config.Log4j1ConfigurationConverter \
+  --in log4j.properties --out log4j2.xml
+----
+====
 
-Log4j 1.x has a XMLLayout which is different from the XmlLayout in Log4j 2.
-The log4j-1.2-api module contains a `Log4j1XmlLayout` that produces output in the Log4j 1.x format.
+[#configuration-interpolation]
+==== Interpolation
 
-The Log4j 1.x `SimpleLayout` can be emulated with PatternLayout "%level - %m%n".
+Log4j 1 only supports interpolation using system properties and properties from the `log4j.properties` file using the `$\{foo}` syntax.
+Log4j 2 extended this mechanism, by introducing pluggable xref:manual/lookups.adoc[].
 
-The Log4j 1.x `TTCCLayout` can be emulated with PatternLayout "%r [%t] %p %c %notEmpty{%ndc }- %m%n".
+In order to convert a Log4j 1 configuration file that uses interpolation to a Log4j 2 configuration file, replace all occurrences of `$\{foo}` with `${sys:foo}`.
 
-Both `PatternLayout` and `EnhancedPatternLayout` in Log4j 1.x can be replaced with `PatternLayout` in Log4j 2.
-The log4j-1.2-api module contains two pattern conversions "%ndc" and "%properties" which can be used to emulate "%x" and "%X" in Log4j 1.x PatternLayout ("%x" and %X" in Log4j 2 have a slightly different format).
+[#configuration-appenders]
+==== Appenders
 
-Below are some example configurations for Log4j 1.x and their counterparts in Log4j 2.
+Log4j 2 contains an equivalent for most Log4j 1 appenders:
 
-=== Sample 1 - Migrating a simple Console Appender configuration
+.Log4j 2 equivalents of Log4j 1 appenders
+[cols="1m,1m,2"]
+|===
+| Log4j 1 appender | Log4j 2 equivalent | Notes
 
-Log4j 1.x XML configuration
+| {log4j1-url}/apidocs/org/apache/log4j/AsyncAppender.html[org.apache.log4j.AsyncAppender]
+| xref:manual/appenders.adoc#AsyncAppender[Async]
+|
 
-[,xml]
-----
-<!DOCTYPE log4j:configuration PUBLIC "-//APACHE//DTD LOG4J 1.2//EN" "log4j.dtd">
-<log4j:configuration xmlns:log4j='http://jakarta.apache.org/log4j/'>
-  <appender name="STDOUT" class="org.apache.log4j.ConsoleAppender">
-    <layout class="org.apache.log4j.PatternLayout">
-      <param name="ConversionPattern" value="%d %-5p [%t] %C{2} (%F:%L) - %m%n"/>
-    </layout>
-  </appender>
-  <category name="org.apache.log4j.xml">
-    <priority value="info" />
-  </category>
-  <Root>
-    <priority value ="debug" />
-    <appender-ref ref="STDOUT" />
-  </Root>
-</log4j:configuration>
-----
+| {log4j1-url}/apidocs/org/apache/log4j/ConsoleAppender.html[org.apache.log4j.ConsoleAppender]
+| xref:manual/appenders.adoc#ConsoleAppender[Console]
+|
 
-Log4j 2 XML configuration
+| {log4j1-url}/apidocs/org/apache/log4j/DailyRollingFileAppender.html[org.apache.log4j.DailyRollingFileAppender]
+| xref:manual/appenders.adoc#RollingFileAppender[RollingFile]
+| See <<rolling-file-conversion>>
```

Comment:
Link is missing a label: 
![image](https://github.com/apache/logging-log4j2/assets/72137/b0049e9e-b152-4d2d-8a21-030ae2b4a32d)

Reply:
Fixed in 4b38c891c6a268e27ad8102e2a49004a4877524e